src/Entities/Player.cpp

```diff
@@ -941,10 +941,29 @@ void cPlayer::ApplyArmorDamage(int DamageBlocked)
 	{
 		ArmorDamage = 1;
 	}
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 0, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 1, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 2, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 3, ArmorDamage);
+
+	for (int i = 0; i < 4; i++)
+	{
+		// (60 + (40 / (UnbreakingLevel + 1)))% chance the armor takes damage
+		// Ref: https://minecraft.gamepedia.com/Enchanting#Unbreaking
+		unsigned int UnbreakingLevel = m_Inventory.GetArmorSlot(i).m_Enchantments.GetLevel(cEnchantments::enchUnbreaking);
+		double chance = (0.6 + (0.4 / (UnbreakingLevel + 1)));
+
+		// When durability is reduced by multiple points
+		// Unbreaking is applied for each point of reduction.
+		for (int dp = 1; dp <= ArmorDamage; dp++)
+		{
+			if (!GetRandomProvider().RandBool(chance))
```

Comment:
Another way to do this would be with [`std::binomial_distribution`](http://en.cppreference.com/w/cpp/numeric/random/binomial_distribution).

```cpp
std::binomial_distribution<short> Dist(ArmorDamage, chance);
auto ReducedDamage = Dist(GetRandomPovider().Engine());
m_Inventory.DamageItem(cInventory::invArmorOffset + i, ReducedDamage)
```